src/overlays/actors/ovl_Bg_Spot15_Rrbox/z_bg_spot15_rrbox.c

```diff
@@ -21,45 +39,343 @@ const ActorInit Bg_Spot15_Rrbox_InitVars = {
+        func_808B44B8(this, globalCtx);
+        this->dyna.actor.posRot.pos = D_808B45C4[D_808B4590];
+        D_808B4590++;
+    } else {
+        func_808B4084(this, globalCtx);
+    }
+    osSyncPrintf("(spot15 ロンロン木箱)(arg_data 0x%04x)\n", this->dyna.actor.params);
+}
+
+void BgSpot15Rrbox_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    BgSpot15Rrbox* this = THIS;
+    DynaPolyInfo_Free(globalCtx, &globalCtx->colCtx.dyna, this->dyna.dynaPolyId);
+    D_808B4590 = 0;
+}
+
+s32 func_808B3CA0(BgSpot15Rrbox* this, GlobalContext* globalCtx, s32 arg2) {
+    f32 chkDist = 0.0f;
+    Vec3f sp38;
+    Vec3f sp2C;
+
+    func_808B3A34(this);
+
+    sp2C.x = D_808B45DC[arg2].x * (this->dyna.actor.scale.x * 10.0f);
+    sp2C.y = D_808B45DC[arg2].y * (this->dyna.actor.scale.y * 10.0f);
+    sp2C.z = D_808B45DC[arg2].z * (this->dyna.actor.scale.z * 10.0f);
+
+    func_808B39E8(&sp38, &sp2C, this->unk_16C, this->unk_170);
+
+    sp38.x += this->dyna.actor.posRot.pos.x;
+    sp38.y += this->dyna.actor.pos4.y;
+    sp38.z += this->dyna.actor.posRot.pos.z;
+
+    this->dyna.actor.groundY = func_8003CA64(&globalCtx->colCtx, &this->dyna.actor.floorPoly, &this->bgId,
+                                             &this->dyna.actor, &sp38, chkDist);
+
+    if (-0.001f <= (this->dyna.actor.groundY - this->dyna.actor.posRot.pos.y)) {
+        this->dyna.actor.posRot.pos.y = this->dyna.actor.groundY;
+        return 1;
+    }
+    return 0;
+}
+
+f32 func_808B3DDC(BgSpot15Rrbox* this, GlobalContext* globalCtx) {
+    int i;
+    Vec3f tempVector2;
+    Vec3f tempVector1;
+    Actor* actor = &this->dyna.actor;
+    f32 yIntersect;
+    f32 returnValue = -32000.0f;
+    s32 bgId;
+
+    func_808B3A34(this);
+    for (i = 0; i < ARRAY_COUNT(D_808B45DC); i++) {
+        tempVector1.x = D_808B45DC[i].x * (actor->scale.x * 10.0f);
+        tempVector1.y = D_808B45DC[i].y * (actor->scale.y * 10.0f);
+        tempVector1.z = D_808B45DC[i].z * (actor->scale.z * 10.0f);
+
+        func_808B39E8(&tempVector2, &tempVector1, this->unk_16C, this->unk_170);
+
+        tempVector2.x += actor->posRot.pos.x;
+        tempVector2.y += actor->pos4.y;
+        tempVector2.z += actor->posRot.pos.z;
+
+        yIntersect = func_8003CA64(&globalCtx->colCtx, &actor->floorPoly, &bgId, actor, &tempVector2, 0);
+
+        if (returnValue < yIntersect) {
+            returnValue = yIntersect;
+            this->bgId = bgId;
+        }
+    }
+    return returnValue;
+}
+
+s32 func_808B3F58(BgSpot15Rrbox* this, GlobalContext* globalCtx) {
+    if (func_808B3CA0(this, globalCtx, 0) != 0) {
+        return 1;
+    }
+    if (func_808B3CA0(this, globalCtx, 1) != 0) {
+        return 1;
+    }
+    if (func_808B3CA0(this, globalCtx, 2) != 0) {
+        return 1;
+    }
+    if (func_808B3CA0(this, globalCtx, 3) != 0) {
+        return 1;
+    }
+    if (func_808B3CA0(this, globalCtx, 4) != 0) {
+        return 1;
+    }
+    return 0;
```

Comment:
can you or `||` these?
can you just return the conditional result?

Reply:
I tried a few different approaches to these lines and couldn't get away from having 5 separate if statements.  I tried oring them together as you suggested and it didn't match any more.  A for loop didn't work either.